src/components/course/data/utils.jsx

```diff
@@ -146,18 +146,15 @@ export const findEnterpriseOfferForCourse = ({
     return undefined;
   }
 
-  return enterpriseOffers.find((enterpriseOffer) => {
+  const applicableEnterpriseOffers = enterpriseOffers.filter((enterpriseOffer) => {
```

Comment:
Rather than returning the first enterprise offer that matches the catalog(s) associated with the course being viewed, we instead will intentionally prioritize offers with no bookings limit first, then offers with a user bookings limit, and then offers with a global bookings limit.